compiler/coreSyn/TrieMap.hs

```diff
@@ -1120,16 +1125,17 @@ type RigMap = VarEnv Rig
 -- of these data types have binding forms.
 
 lkBndr :: CmEnv -> Var -> BndrMap a -> Maybe a
-lkBndr env v (BndrMap rm tymap) = do
-  w <- lookupVarEnv rm v
-  guard (fromMaybe Omega (varWeightMaybe v) == w)
-  lkG (D env (varType v)) tymap
+lkBndr env v (BndrMap tymap) = do
+  (a, w) <- lkG (D env (varType v)) tymap
+  guard (w == varWeightDef v)
```

Comment:
I'm amused at how you flipped the order of the operands of `(==)`.

Some say I'm easily amused.